### PR TITLE
Update GitLab example for 1.9

### PR DESCRIPTION
**Pull request by pleia2**
While updating links and images to 1.9, also clarified how to add a credential
to Jenkins, make clear that the /etc/docker.tar.gz on all nodes is required for
authentication when launching the image and remove the deprecated -e flag from
the docker login command in the Jenkinsfile.

### CODE REVIEW

File: gitlab/1.9/README.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+- An available hostname configured to point to the public agent(s) where Marathon-lb is running (e.g. `gitlab-test.mesosphere.com`)
+- Either set up appropriate certificates on each of your hosts, or configure each of your hosts to use your hostname (i.e. `gitlab-test.mesosphere.com`) as a [Docker insecure registry](https://docs.docker.com/registry/insecure/).
+You will need to do this for the Jenkins agent too using the [Advanced Configuration instructions](https://docs.mesosphere.com/1.9/usage/service-guides/jenkins/advanced-configuration/).
+- Ports 22222 and 50000 opened on the public agent where Marathon-lb is running. If you're using an ELB or similar in front of the public agent, make sure it's listening on those ports too.
+
+## Setting up GitLab
+
+1. Before starting, identify the hostname of a private agent that you'd like to install GitLab to. Typically this will be one that has an EBS volume or similar mounted, that you are regularly snapshotting or have set up some other sort of backup solution on. You can pick one of these by visiting the "Nodes" page on your cluster and choosing a private node. We'll use `10.0.0.134` for this example.
```
Comment (OhRobin) [inline gitlab/1.9/README.md]: do you mean `10.0.0.133` as shown in the screenshot?

Comment (pleia2) [inline gitlab/1.9/README.md]: I do indeed! I'll update the PR.

File: gitlab/1.9/README.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+- [Setting up the Repository](#setting-up-the-repository)
+- [Set up the Pipeline](#set-up-the-pipeline)
+- [Resources](#resources)
+
+## Prerequisites
+
+- DC/OS 1.9 or later
+- [Marathon-lb](https://dcos.io/docs/1.9/usage/service-discovery/marathon-lb/usage/) must be installed and running
+- [Jenkins](https://docs.mesosphere.com/1.9/usage/service-guides/jenkins/) must be installed and running
+- An available hostname configured to point to the public agent(s) where Marathon-lb is running (e.g. `gitlab-test.mesosphere.com`)
```
Comment (OhRobin) [inline gitlab/1.9/README.md]: I'm following the example here using gitlab 9 1 0 version. And I was wondering if you can help clarify having hostname configured? I'm having trouble in this step

File: gitlab/1.9/README.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+- Either set up appropriate certificates on each of your hosts, or configure each of your hosts to use your hostname (i.e. `gitlab-test.mesosphere.com`) as a [Docker insecure registry](https://docs.docker.com/registry/insecure/).
```
Comment (OhRobin) [inline gitlab/1.9/README.md]: Would love clarification here on installation. Do you recommend using docker registry dcos package?

File: gitlab/1.9/README.md (hunk 2 above)
Comment (pleia2) [inline gitlab/1.9/README.md]: This just means the public address of where you have marathon-lb running :)

A suggested DNS name is used if you want to formally set this up (we don't often do this for examples, but here we are), but you can just use whatever DNS is automatically assigned by your cloud provider or network.

File: gitlab/1.9/README.md (hunk 3 above)
Comment (pleia2) [inline gitlab/1.9/README.md]: No, GitLab itself is a registry. By default it's insecure. It's telling you to either set up SSL certificates (making it secure), or to tell Docker to trust your specific insecure GitLab registry by following instructions in the link.

**Comment by ssk2** [comment]
The GItLab portion of this demo looks good to me. I think we need more instruction around credentials. Specifically:
+ How to set up Marathon-lb with Lets Encrypt (do we have this documented anywhere? Worth calling out explicitly - it'll avoid the pain with either distributing certs or setting an unrestricted registry).
+ Double check the Jenkins portion is accurate - there have been some changes to the Marathon plugin lately. I'd suggest asking @colin-msphere to verify this.

**Comment by pleia2** [comment]
Thanks for the feedback, @ssk2. I'm wondering how much of this we need to pile into this commit, and how  much is reasonable to break out to future tasks. It would help me to get the basic 1.9 update out the door to close out that work item, then follow up with improvements.

Coming up with a general policy around using Let's Encrypt on some other things our team is working on is also on my radar (there are a few things to consider, including not abusing the service and making sure we have instructions that are both clear and as simple as possible).

**Comment by ssk2** [comment]
I'd suggest double checking the Jenkins portion - but otherwise I'm fine with deferring work around Marathon-lb docs to a future change.

**Comment by ssk2** [comment]
I'd suggest double checking the Jenkins portion - but otherwise I'm fine with deferring work around Marathon-lb docs to a future change.

**Comment by lambertjosh** [comment]
How are we doing on this one?

**Pull request merged**